app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -52,6 +52,10 @@ XrResult OpenXRInputSource::Initialize()
       if (mapping.systemFilter && strcmp(mapping.systemFilter, mSystemProperties.systemName) != 0) {
         continue;
       }
+      bool systemIs6DoF = mSystemProperties.trackingProperties.positionTracking == XR_TRUE;
```

Comment:
If I understand this correctly, placing this code at `Initialize()` means that we will only do this check at the beginning.

Would it make sense to add both `Hvr6DOF` and `Hvr3DOF` to the `mMappings` vector?

We could then move this check to the place where the vector is used (`UpdateInteractionProfile`, I think). Would this make sense?

Reply:
So, I guess you suggest that to allow selecting one or the other dynamically. 

From the OpenXR POV you're totally right, we should be able to register different profiles with different suggested bindings for a given system. The problem is that our OpenXRInputMappings are a bit of a mixture of things, they do indeed contain some OpenXR suggested bindings but also things like the filenames with the controller 3D models.

The 3D models of the controllers are loaded just once in BrowserWorld::InitializeJava(), so even if we register 3DoF and 6DoF profiles we wouldn't be able to properly switch between them at runtime. I agree this might be a desirable feature but it's something that we can't do right now.

Reply:
Ok, that sounds reasonable.